### ISSUE TITLE
Should cast_nullable_to_non_nullable warn if dynamic is casted to non-nullable?

### ISSUE THREAD

**Issue by goderbauer** [dart-lang/linter#2828]
```dart
String foo(dynamic foo) {
  return foo as String;
}
```

I expected a `cast_nullable_to_non_nullable` warning on the line above reminding me to either cast to `String?` or to add an explicit non-null check. Unfortunately, the lint doesn't trigger today. I would argue that the code above is also hiding "a null check and most of the time it is not what is expected." (to cite from https://dart-lang.github.io/linter/lints/cast_nullable_to_non_nullable.html).

/cc @pq @a14n @gspencergoog

**Comment by a14n** [comment]
See https://github.com/dart-lang/linter/issues/2235#issuecomment-696823493

**Comment by pq** [comment]
@srawlins may have some thoughts here too.

**Comment by SuporteSysrs01** [comment]
I have a really strong preference to it doing so in my case, specially with implicit casts.
I've made several comments in the dart-lang/language#1022 issue explaining my use cases.

**Comment by jonaslundquist** [comment]
I just want to +1 this idea very much! several production issues have been caused by assuming non-nullable when casting from dynamic, especially using `[]` operator on `Map`
